### PR TITLE
Aggregate some operator runtime metrics.

### PR DESCRIPTION
**Pull request by spershin**
Summary:
Aggregate few operator runtime metrics: "dataSourceWallNanos", "dataSourceLazyWallNanos",
"queuedWallNanos" and "blocked*WallNanos".
This allows us to see the long tail (the worst Driver).

Differential Revision: D44485872

### CODE REVIEW

File: velox/exec/OperatorUtils.cpp [hunk 1]
```diff
@@ -116,6 +117,26 @@ void gatherCopy(
...
+  static const std::array<std::string, 3> metricNames{
+      "dataSourceWallNanos", "dataSourceLazyWallNanos", "queuedWallNanos"};
+  for (const auto& metricName : metricNames) {
+    if (name == metricName) {
+      return true;
+    }
+  }
+
+  // 'blocked*WallNanos'
+  if (name.size() > 16 and strncmp(name.c_str(), "blocked", 7) == 0) {
```
Comment (xiaoxmeng) [inline velox/exec/OperatorUtils.cpp:133]: use regular expression here? like static const boost::regex re("^[a-zA-Z_][a-zA-Z0-9_]*$");? Thanks!

File: velox/exec/OperatorUtils.cpp [hunk 2]
```diff
@@ -116,6 +117,26 @@ void gatherCopy(
         target, targetIndex, count, sources, sourceIndices, sourceChannel);
   }
 }
+
+// We want to aggregate some operator runtime metrics per operator rather than
+// per event. This function returns true for such metrics.
+bool shouldAggregateRuntimeMetric(const std::string& name) {
+  static const std::array<std::string, 3> metricNames{
```
Comment (xiaoxmeng) [inline velox/exec/OperatorUtils.cpp:124]: Why not use folly::F14FastSet here?

**Review by xiaoxmeng — COMMENTED** [review]
@spershin Looks good % minors. Thanks!

File: velox/exec/OperatorUtils.cpp (hunk 1 above)
Comment (spershin) [reply velox/exec/OperatorUtils.cpp:133]: I believe it is not worth using re here. Using strncmp is lighter.

File: velox/exec/OperatorUtils.cpp (hunk 2 above)
Comment (spershin) [reply velox/exec/OperatorUtils.cpp:124]: Good point, will change. :)

File: velox/exec/OperatorUtils.cpp (hunk 1 above)
Comment (xiaoxmeng) [reply velox/exec/OperatorUtils.cpp:129]: But such a comparison is a bit fragile? re expression is actually easy to understand?

**Review by xiaoxmeng — APPROVED** [review]
@spershin thanks for the update!

**Pull request closed without merging**